servercom/servercom.go

```diff
@@ -0,0 +1,138 @@
+package serverCom
+
+import (
+	"fmt"
+	mm "github.com/mattermost/platform/model"
+)
+/*
+ServerCom acts as a mitigator between the frontend and the mattermost model API.
+*/
+type ServerCom struct {
+	Client mm.Client
+	Team mm.Team
+	Channel *mm.Channel
+}
+
+/*
+Startup accepts the url and login credentials for a user, and returns a new serverCom struct.
+*/
+func Startup(url string, username string, password string) ServerCom {
+	ServerCom := ServerCom{Client: *mm.NewClient(url)}
+	_, err := ServerCom.Client.Login(username, password)
+	if err != nil {
+		fmt.Println(err)
+	}
+	return ServerCom
+}
+
+/*
+Connected returns true if the client has a login Authentication Token.
+*/
+func (sc *ServerCom) Connected() bool {
+	if sc.Client.AuthToken != "" {
+		return true
+	}
+	return false
+}
+
+/*
+SetTeam set ServerCom's client team ID using the team name.
+*/
+func (sc *ServerCom) SetTeam(teamName string) {
+	team, err := sc.Client.GetTeamByName(teamName)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	sc.Team = *team.Data.(*mm.Team)
+	sc.Client.SetTeamId(sc.Team.Id)
+	return 
+}
+
+func (sc *ServerCom) GetTeams() map[string]*mm.Team {
+	teamListResult, teamListAppError := sc.Client.GetAllTeamListings()
+	teamMap := teamListResult.Data.(map[string]*mm.Team)
+	if teamListAppError != nil {
+		fmt.Println(teamListAppError)
+		return nil
+	}
+	return teamMap
+}
+
+func (sc *ServerCom) PrintTeams() {
+	fmt.Println("Teams:")
+	for name, value := range sc.GetTeams() {
+		fmt.Println("\tName:", value.Name, "TeamID:", name)
+	}
+}
+
+func (sc *ServerCom) SetChannel(channelName string) {
+	channelResult, err := sc.Client.GetChannelByName(channelName)
+	if err != nil {
+		fmt.Println(err)
+	}
+	sc.Channel = channelResult.Data.(*mm.Channel)
+	return 
+}
+
+//GetChannels returns a map with all availible channels in team.
+//**Must have successfully run SetTeam()**
+func (sc *ServerCom) GetChannels() *mm.ChannelList {
```

Comment:
Maybe this should explicitly check whether `SetTeam` has been called. If not, you could return a custom error using `fmt.Errorf()`